### ISSUE TITLE
WSL build instructions need updating

### ISSUE THREAD

**Issue by bhaller** [MesserLab/SLiM#467]
The instructions for building SLiM under the WSL that are in the manual are rather old and obsolete, I gather.  It would be great if somebody could update them.  If you, dear reader, want to get into that, let me know and we'll figure out how to proceed.  Tagging folks who might be interested: @bernard-kim @bryce-carson @rdinnager.

**Comment by bhaller** [comment]
Also tagging @chazhyseni!

**Comment by bhaller** [comment]
It would really be nice to get this updated for the SLiM 5 release (early to mid April).  I think there's a fair number of people who use SLiM under the WSL.  Any takers?

**Comment by chazhyseni** [comment]
I'll find a bit of time. Will have a look at some point in the next couple of weeks.

**Comment by bhaller** [comment]
> I'll find a bit of time. Will have a look at some point in the next couple of weeks.

Awesome, thanks @chazhyseni!  The docs are in Pages, a macOS app format.  I could export them to Word.  You could do tracked changes in either format and I could merge them back in.  Or you could just write something new, if things have changed that much.  Let me know how you'd like to proceed!  :->

**Comment by bryce-carson** [comment]
As a friendly note, now that graphical Linux applications running atop WSL 2 have native support in Windows 11, it is much easier to instruct users to follow the relevant Microsoft documentation and then tell them to follow the build instructions for Ubuntu.

It usually comes down to this on a Windows 11 machine now:
- press start
- type "Ubuntu'
- "install Ubuntu" from the windows store, following the search result seen in your start menu
- open Ubuntu
- setup Ubuntu, as necessary
- download and build SLiM

If there are any extra steps for .desktop files to be automatically recognized by Windows and integrated into the Windows Shell, I don't know what they are, but it's pretty breezy to use Linux graphical applications on Windows these days. It should be a much better UX now.

Windows 10 may be behind, but I think that's technically EOL already? I don't know.

**Comment by bhaller** [comment]
@bryce-carson Aha, I see.  Well, I'm happy to only provide instructions for Windows 11, if that is what people ought to be on now.  It could say "For instructions that work on Windows 10, please see the version of this manual provided for SLiM 4.3, which can be obtained from the Releases area of SLiM's GitHub repository, at XXXXX."  @chazhyseni Please see what you think of this.  If there are any complications beyond what Bryce has outlined, or any steps that seem to need a screenshot to be clear, please provide that kind of stuff to me.  If it is really this simple now, then I can write it up myself, just basically copy/pasting what Bryce wrote above with a bit of embellishment.  :->  But I'm not a Windows person so I can't walk through these steps myself to confirm the procedure and take any necessary screenshots, so I do still need your help with that, Chaz.  Thanks!  (Boy, I'm looking forward to deleting the pages of cruft about the WSL that are there now!  :->)

**Comment by bhaller** [comment]
Hmm.  It's an idea worth considering.  Where are you envisioning that these instructions would live, then?  People should not need to go to the source code archive, or to GitHub at all, to get install instructions; many users are not programmers at all, have never used GitHub, etc., so there's a barrier there for many people (a barrier that might seem odd to us, who are used to GitHub etc., but it's a real barrier for many new users).  On the other hand, install instructions should be tightly associated with a particular release, and people should be able to find the install instructions for previous releases.  (Those instructions would not necessarily _work_ any more, of course, but it should be possible to at least look at them and see what has changed from version to version.)  And install instructions often involve embedded graphics like screenshots, so a text-only file does not suffice.  To me, with those constraints, the logical place is the manual; and usually the manual is where software projects put install instructions, so it's where people will expect to find them.  So... I'm leaning towards leaving things as they are.  But do you feel strongly otherwise?  Do we in fact want/need to change the install instructions at times other than at the moment of a new release?  Why would that be useful?

**Comment by bryce-carson** [comment]
> Do we in fact want/need to change the install instructions at times other than at the moment of a new release?  Why would that be useful?

I'd hope we wouldn't need to update them, but in the event we do it would be good if people would see these updated instructions before the outdated instructions.

Perhaps we can distribute the installation instructions, in whatever format is best for them (an HTML document? A PDF?), alongside the manual in a archive file (ZIP?).

I recall now that the manual does include a little sentence detailing what release it corresponds to and also what the last date the manual was updated, but given the more involved process of updating the manual maybe it would be easier to update the instructions for installing the software separately.

Mostly, I am thinking that a different UX like an HMTL document would allow us to provide a richer experience for navigating and also foe reporting installation issues.

Users could graphically select their operating system and then be presented with the instructions and any caveats for just that OS. Hyperlinks would be nice (though they do exist in PDF) to have.

Code snippets that can be copied more easily and reliably than drag and select would also be nice (especially given PDF isn't the best with this sometimes).

It's not a need, no, but "for want to experiment" for users benefit it might be nice to have a separate document.

We could even provide *rich diffs* in that document, like gitbook.

It's only a thought, I just think the installation section in the manual is a little cramped visually and it might be nice to offer something "fresher".

**Comment by bhaller** [comment]
Yes, that does sound nicer.  It seems to me that the install instructions could actually be online, in fact.  I can put them up at benhaller.com.  There can be a separate HTML main page (under a different folder on the web server, with different sub-pages and images) for each release, so that people can find older releases; and then they can be updated live on the web if needed.  The manual can still have a chapter 2 on installation, but it would pretty much just talk about the general strategy and link to the online instructions for the details.  (Perhaps the "post-install issues" content and that sort of thing, other than actual installation, would remain in the manual?)  I agree that hyperlinks would be particularly useful for this content, since there are "choose your own adventure" style decisions to be made on OS, build from sources vs. use an installer, etc.  I'm not sure that I'll have time to do this for the SLiM 5 release, though; and I'd like a little time to mull it over.  So.  Let's think about it, and if it still seems like a good idea, you or I can open a new issue on it – it's orthogonal to this WSL issue, in any case.  :->  Thanks for the idea!

**Comment by bryce-carson** [comment]
Indeed. 😄

**Comment by bhaller** [comment]
SLiM 5 is scheduled for release this Friday, April 18th, unless something goes wrong.  It'd be great if someone has time for this issue.  :->

**Comment by bhaller** [comment]
Hi @chazhyseni!  No worries, life is crazy these days.  Happy to do this any time you want.  You could work from the text in the SLiM 5.0 manual; I imagine you will change it more or less completely anyway.  It sounds like the new section on the GSL can probably be very short and simple.  The original doc is in Pages format, which is a macOS text editor; I can send that to you if you're a macOS user.  But if you just write new content, in Word or plain text or whatever you like, I can merge it in.  :->  Thanks!
